cutil/logger/logger.go

```diff
@@ -137,6 +125,20 @@ func Warning(format string, a ...interface{}) {
 	}
 }
 
+func extractLoggerArgs(format string, a ...interface{}) ([]interface{}, io.Writer) {
```

Comment:
Yes, and the reason we should have this right is, this is a logger package and will be used by almost every package and any one writing a new package shouldn't have to go look into the logger package implementation to see, how, stuff works. API should tell him what needs to be passed in to the calls.